neuralpp/symbolic/z3_expression.py

```diff
@@ -413,25 +427,25 @@ def dict(self) -> Dict[str, Any]:
     def syntactic_eq(self, other) -> bool:
         return False  # why do we need to compare two Z3ConjunctiveClause?
 
-    def __init__(self, z3_solver: z3.Solver, value_dict: Dict[str, Any] | None = None):
+    def __init__(self, z3_solver: z3.Solver = z3.Solver(), value_dict: Dict[str, Any] | None = None):
```

Comment:
We aren't going to mutate `z3_solver` in `Z3SolverExpression`, right? If so, this is fine; otherwise, we should be careful about having a mutable object as the default argument [as it can lead to unexpected outcomes](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)